src/test/java/com/amazonaws/encryptionsdk/NewCompatTest.java

```diff
@@ -0,0 +1,296 @@
+
+    public NewCompatTest(final String testName, TestCase testCase) {
+        this.testName = testName;
+        this.testCase = testCase;
+    }
+
+    @Test
+    public void decrypt() {
+        AwsCrypto crypto = new AwsCrypto();
+        byte[] plaintext = crypto.decryptData(testCase.mkp, cachedData.get(testCase.ciphertextPath)).getResult();
+        final byte[] expectedPlaintext = cachedData.get(testCase.plaintextPath);
+
+        Assert.assertArrayEquals(expectedPlaintext, plaintext);
+    }
+
+    @Parameterized.Parameters(name="Compatability Test: {0}")
+    @SuppressWarnings("unchecked")
+    public static Collection<Object[]> data() throws Exception {
+        final String zipPath = System.getProperty("testVectorZip");
+        if (zipPath == null) {
+            return Collections.emptyList();
+        }
+
+        final JarURLConnection jarConnection = (JarURLConnection) new URL("jar:" + zipPath + "!/").openConnection();
+
+        try (JarFile jar = jarConnection.getJarFile()) {
+            final ObjectMapper mapper = new ObjectMapper();
+            final Map<String, Object> manifest = readJsonMapFromJar(jar, "manifest.json");
+
+            final Map<String, Object> metaData = (Map<String, Object>) manifest.get("manifest");
+
+            // We only support "awses-decrypt" type manifests right now
+            if (!"awses-decrypt".equals(metaData.get("type"))) {
+                throw new IllegalArgumentException("Unsupported manifest type: " + metaData.get("type"));
+            }
+
+            if (!Integer.valueOf(1).equals(metaData.get("version"))) {
+                throw new IllegalArgumentException("Unsupported manifest version: " + metaData.get("version"));
+            }
+
+            final Map<String, KeyEntry> keys = parseKeyManifest(readJsonMapFromJar(jar, (String) manifest.get("keys")));
+
+            final KmsMasterKeyProvider kmsProv = KmsMasterKeyProvider
+                                                         .builder()
+                                                         .withCredentials(new DefaultAWSCredentialsProviderChain())
+                                                         .build();
+
+            List<Object[]> testCases = new ArrayList<>();
+            for (Map.Entry<String, Map<String, Object>> testEntry :
+                    ((Map<String, Map<String, Object>>) manifest.get("tests")).entrySet()) {
+                testCases.add(new Object[]{testEntry.getKey(),
+                        parseTest(testEntry.getKey(), testEntry.getValue(), keys, jar, kmsProv)});
+            }
+            return testCases;
+        }
+    }
+
+    @AfterClass
+    public static void teardown() {
+        cachedData.clear();
+    }
+
+    private static byte[] readBytesFromJar(JarFile jar, String fileName) throws IOException {
+        try (InputStream is = readFromJar(jar, fileName)) {
+            return IOUtils.toByteArray(is);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
```

Comment:
I don't think this suppression is necessary

Reply:
Needed by a prior iteration of this method. Removed.